node/src/proxy_server/mod.rs

```diff
@@ -851,10 +979,77 @@ pub trait IBCDHelper {
         msg: InboundClientData,
         retire_stream_key: bool,
     ) -> Result<(), String>;
+
+    fn request_route_and_transmit(
+        &self,
+        tth_args: TryTransmitToHopperArgs,
+        route_source: Recipient<RouteQueryMessage>,
+        proxy_server_sub: Recipient<RouteResultMessage>,
+    );
+}
+
+trait RouteQueryResponseResolver: Send {
+    fn resolve_message(
+        &self,
+        args: TryTransmitToHopperArgs,
+        proxy_server_sub: Recipient<RouteResultMessage>,
+        route_result: Result<Option<RouteQueryResponse>, MailboxError>,
+    );
+}
+struct RouteQueryResponseResolverReal {}
+
+impl RouteQueryResponseResolver for RouteQueryResponseResolverReal {
+    fn resolve_message(
+        &self,
+        args: TryTransmitToHopperArgs,
+        proxy_server_sub: Recipient<RouteResultMessage>,
+        route_result: Result<Option<RouteQueryResponse>, MailboxError>,
```

Comment:
This `MailboxError` type seems as though it couples this method (and trait) a little too tightly to Actix. The only thing this method does with the `MailboxError` is convert it into a `String`. Could you have the caller of this method (which is already coupled to Actix error handling) do that conversion and pass in a `Result<Option<RouteQueryResponse>, String>`?

Reply:
This type `Result<Option<RouteQueryResponse>, MailboxError>` is tightly coupled with the futures `.then()` function.
The closure indented for the `.then()` function receives this type as an argument due to this reason we see this type in different places within our code base. 

Do we need to type cast this type into a different one?